riposte-core/src/test/java/com/nike/riposte/server/handler/RequestInfoSetterHandlerTest.java

```diff
@@ -93,4 +121,114 @@ public void doChannelRead_checks_for_fully_send_responses_but_does_nothing_else_
         assertThat(result).isEqualTo(PipelineContinuationBehavior.CONTINUE);
     }
 
+    @Test
+    public void doChannelRead_returns_do_not_fire_when_null_state() {
+        // given
+        doReturn(true).when(stateMock).isResponseSendingLastChunkSent();
+
+        // when
+        PipelineContinuationBehavior result = handler.doChannelRead(ctxMock, httpContent);
+
+        // then
+        assertThat(result).isEqualTo(PipelineContinuationBehavior.DO_NOT_FIRE_CONTINUE_EVENT);
+    }
+
+    @Test
+    public void doChannelRead_throws_exception_when_no_request_info_when_HttpContent_message() {
+        // given
+        doReturn(null).when(stateMock).getRequestInfo();
+        Exception thrownException = null;
+
+        // when
+        try {
+            handler.doChannelRead(ctxMock, httpContent);
+            fail();
+        } catch (Exception e) {
+            thrownException = e;
+        }
+
+        // then
+        assertThat(thrownException).isExactlyInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    public void doChannelRead_does_not_throw_exception_when_exceeding_global_max_size_when_request_validation_is_turned_off() {
+        // given
+        doReturn(Optional.of(0)).when(endpointMock).maxRequestSizeInBytesOverride();
+        doReturn(100).when(requestInfo).addContentChunk(anyObject());
+
+        // when
+        PipelineContinuationBehavior result = handler.doChannelRead(ctxMock, httpContent);
+
+        // then
+        assertThat(result).isEqualTo(PipelineContinuationBehavior.CONTINUE);
+    }
+
+    @Test
+    public void doChannelRead_allows_endpoints_to_override_max_request_size_setting_lower() {
+        // given
+        //handler is configured for 10
```

Comment:
Reference `maxRequestSizeInBytes` in the test so you don't have to put these comments here and also removes the potential for accidental test breakage or worse, accidental regression in that the tests don't catch.